lib/internal/modules/esm/resolve.js

```diff
@@ -584,7 +527,7 @@ function packageResolve(specifier, base, conditions) {
   let lastPath;
   do {
     const stat = tryStatSync(
-      StringPrototypeSlice(packageJSONPath, 0, packageJSONPath.length - 13));
+      dirname(packageJSONPath));
```

Comment:
I would actually prefer we stick with slicing just because it's quite well-defined what string manipulation steps are involved.